File: Algorithms/String_Algorithms/LPS/longest_palindromic_substring.py
def longestPalSubstr(string): 
    """
    This function prints the longest palindrome substring (LPS) 
    of string. It also returns the length of the longest palindrome 
    :param string: The string to check for palindrome
    """

    maximumLength = 1

    start = 0
    length = len(string) 

    low = 0
    high = 0

    # One by one consider every character as center point of 
    # even and length palindromes 
    for i in range(1, length): 
        # Find the longest even length palindrome with center 
        # points as i-1 and i. 
        low = i - 1
        high = i 
        while((low >= 0) and (high < length) and (string[low] == string[high])): 
            if (high - low + 1 > maximumLength): 
                start = low 
                maximumLength = high - low + 1
            low -= 1
            high += 1

        # Find the longest odd length palindrome with center 
        # point as i 
        low = i - 1
        high = i + 1
        while ((low >= 0) and (high < length) and (string[low] == string[high])): 
            if high - low + 1 > maximumLength: 
                start = low 
                maximumLength = high - low + 1
            low -= 1
            high += 1

    print("Longest palindrome substring is: {}".format(string[start:start + maximumLength]))

# Driver program to test above functions 
string = "anaqwertyuioppoiuytrewqksl"
longestPalSubstr(string)

File: Algorithms/String-Algorithm/LPS/longest_palindromic_substring.c
#include <stdio.h> 
#include <string.h> 
void print(char* str, int low, int high) 
{ 
    for (int i = low; i <= high; ++i) 
        printf("%c", str[i]); 
} 
  

void longestSubstr(char* str) 
{ 
    int maxLength = 1; 
    int start = 0; 
    int len = strlen(str); 
    int low, high;
    for (int i = 1; i < len; ++i) { 
        low = i - 1; 
        high = i; 
        while (low >= 0 && high < len 
               && str[low] == str[high]) { 
            if (high - low + 1 > maxLength) { 
                start = low; 
                maxLength = high - low + 1; 
            } 
            --low; 
            ++high; 
        } 
        low = i - 1; 
        high = i + 1; 
        while (low >= 0 && high < len 
               && str[low] == str[high]) { 
            if (high - low + 1 > maxLength) { 
                start = low; 
                maxLength = high - low + 1; 
            } 
            --low; 
            ++high; 
        } 
    } 
    printf("Longest palindrome substring is: "); 
    print(str, start, start + maxLength - 1); 
} 
  
// Driver program to test above functions 
int main() 
{ 
    char s[1000], i;
    int n;
    printf("enter the length of string \n");
    scanf("%d", &n);
    printf("Please enter your string \n");
    for(i = 0; i < n; i++){
        scanf("%c", &s[i]);
    }
    longestSubstr(s); 
    return 0; 
} 